coreTable/CoreTableUtils.ts

```diff
@@ -158,11 +158,63 @@ export const makeRowFromColumnStore = (
     return row
 }
 
-function isNotInvalidOrEmptyCell(value: any) {
+function isNotInvalidOrEmptyCell<K>(
+    value: K
+): value is Exclude<K, InvalidCell | undefined> {
     return value !== undefined && !(value instanceof InvalidCell)
 }
 
-export function interpolateColumnsWithTolerance(
+export function linearInterpolation(
+    valuesSortedByTimeAsc: (number | InvalidCell)[],
+    timesAsc: Time[],
+    start = 0,
```

Comment:
@MarcelGerber I haven't looked myself, but is there any way we could share more code with the other interpolation? Or is it easier to have 2 totally separate methods?

@danielgavrilov what do you think of this pattern of passing in the starts and ends? It does ensure interpolation is O(N) (assuming things are passed in sorted), but also feels a little brittle in that it may be premature optimization?

Not entirely sure.

It feels like if we had fast groupBy, we wouldn't need it, but not sure if we could get that pattern fast enough?

Anyway, just thinking out loud.

Reply:
Seems fine to me, and probably as optimized as we can make this? 

The groupBy class/method would be cleaner to use but probably hard to achieve the same performance since we can't make the assumption in it that the array will stay the same, some rows might be inserted into a group by some transform, so would need to chunk the columnStore and create copies.